pkg/rancher-desktop/backend/wsl.ts

```diff
@@ -735,14 +735,16 @@ export default class WSLBackend extends events.EventEmitter implements VMBackend
 
       await this.handleUpgrade([OldCredHelperService, OldCredHelperConfd]);
 
-      if (!this.cfg?.experimental.virtualMachine.networkingTunnel) {
-        await this.writeFile('/etc/init.d/vtunnel-peer', SERVICE_VTUNNEL_PEER, 0o755);
-        await this.writeConf('vtunnel-peer', {
-          VTUNNEL_PEER_BINARY: await this.getVtunnelPeerPath(),
-          LOG_DIR:             await this.wslify(paths.logs),
-          CONFIG_PATH:         await this.wslify(getVtunnelConfigPath()),
-        });
-        await this.execCommand('/sbin/rc-update', 'add', 'vtunnel-peer', 'default');
+      await this.writeFile('/etc/init.d/vtunnel-peer', SERVICE_VTUNNEL_PEER, 0o755);
+      await this.writeConf('vtunnel-peer', {
+        VTUNNEL_PEER_BINARY: await this.getVtunnelPeerPath(),
+        LOG_DIR:             await this.wslify(paths.logs),
+        CONFIG_PATH:         await this.wslify(getVtunnelConfigPath()),
+      });
+      await this.execCommand('/sbin/rc-update', 'add', 'vtunnel-peer', 'default');
+
+      if (this.cfg?.experimental.virtualMachine.networkingTunnel) {
+        await this.execCommand('/sbin/rc-update', 'del', 'vtunnel-peer', 'default');
```

Comment:
We should probably have comment about why we do this add-and-delete dance. (Because `del` without `add` returns an error instead of doing nothing.)